app/analytics/collect.ts

```diff
@@ -62,9 +100,11 @@ export function collectRequestHandler(request: Request, env: Env) {
 
     parsedUserAgent.getBrowser().name;
 
-    const { newVisitor, newSession } = checkVisitorSession(
-        request.headers.get("if-modified-since"),
-    );
+    const ifModifiedSince = request.headers.get("if-modified-since");
+    const { newVisitor, newSession } = checkVisitorSession(ifModifiedSince);
+    const modifiedDate = ifModifiedSince
```

Comment:
Nit: you could also write this as

```javascript
const modifiedDate = getNextModifiedDate(ifModifiedSince ? new Date(ifModifiedSince) : null);
```

It's minor but it avoids repeating `getNextModifiedDate`.

---

Note also that if `ifModifiedSince` is not a well-formed date string, passing it to `new Date` will create an "Invalid Date" and likely cause some trouble.

Is it safe to presume browser clients will pass that string properly? _Probably_ ... but it couldn't hurt to be resilient here, or at the very least leave a note this is possible.

Reply:
Sounds good, thanks for the feedback! I made the changes you suggested above, as for malformed dates, I added in the below check in the `getNextModifiedDate` function to default the variable to `null` if the date is malformed:

```
if (current && isNaN(current.getTime())) {
   current = null;
}
```